src/v/config/property.h

```diff
@@ -130,6 +165,105 @@ class property : public base_property {
     constexpr static auto noop_validator = [](const auto&) {
         return std::nullopt;
     };
+
+    std::vector<binding<T>*> _bindings;
```

Comment:
threading the bindings onto a boost intrusive list might simplify the parent/child accounting. similar to the "subscription" object associated with a seastar::abort_source.

Reply:
Nice idea, I've done it - it only saves like 15 lines but they're fiddly detail lines that we're better off without!